src/profile/v100/app_preferences/gateways/saved_gateways.rs

```diff
@@ -293,4 +294,72 @@ mod tests {
             "#,
         )
     }
+
+    #[test]
+    fn deserialize_from_json_with_different_description_marks_both_gateways_as_wellknown(
+    ) {
+        let json = r#"
+        {
+            "current": "https://mainnet.radixdlt.com/",
+            "saved": [
+                {
+                    "network": {
+                        "name": "mainnet",
+                        "id": 1,
+                        "displayDescription": "Mainnet Gateway"
+                    },
+                    "url": "https://mainnet.radixdlt.com/"
+                },
+                {
+                    "network": {
+                        "name": "stokenet",
+                        "id": 2,
+                        "displayDescription": "Stokenet (testnet) Gateway"
+                    },
+                    "url": "https://babylon-stokenet-gateway.radixdlt.com/"
+                }
+            ]
+        }
+        "#;
+
+        let sut = serde_json::from_str::<SUT>(json).unwrap();
+        let is_wellknown_vec = sut
+            .all()
+            .iter()
+            .map(|gateway| gateway.is_wellknown())
+            .collect_vec();
+
+        assert_eq!(vec![true, true], is_wellknown_vec)
+    }
+
+    #[test]
+    fn test_gateways_identification() {
+        let mainnet = Gateway::new(
+            String::from("https://mainnet.radixdlt.com/"),
+            NetworkID::Mainnet,
+        )
+        .unwrap();
+        let mainnet_no_slash = Gateway::new(
+            String::from("https://mainnet.radixdlt.com"),
+            NetworkID::Mainnet,
+        )
+        .unwrap();
+        let other_mainnet = Gateway::new(
+            String::from("https://other-mainnet.radixdlt.com"),
+            NetworkID::Mainnet,
+        )
+        .unwrap();
+        let other_mainnet_http = Gateway::new(
+            String::from("http://other-mainnet.radixdlt.com"),
+            NetworkID::Mainnet,
+        )
+        .unwrap();
+        let gateways_vec =
+            vec![mainnet, mainnet_no_slash, other_mainnet, other_mainnet_http];
+
+        let identified_gateways =
+            Gateways::from_iter(gateways_vec.iter().cloned());
+
+        assert_ne!(gateways_vec.len(), identified_gateways.len())
```

Comment:
I think instead of expecting not equals, better to skip using `gateways_vec` and just assert expected length of `identified_gateways` - with a comment about *why*.